saleor/app/models.py

```diff
@@ -97,10 +98,41 @@ def has_perm(self, perm):
         return perm_value in self.get_permissions()
 
 
+class AppTokenManager(models.Manager):
+    def create_app_token(self, app: App, name: str = "") -> Tuple["AppToken", str]:
+        """Create app token and save the value as hashed password."""
+        token_obj = cast(AppToken, self.model(name=name, app=app))
+        token = generate_token()
+        token_obj.auth_token = make_password(token)
+        token_obj.raw_auth_token_last_4_chars = token[-4:]
+        token_obj.save()
```

Comment:
Shouldn't this use `update_fields`?

Reply:
No, as the instance there is not saved yet. The object in line `104` is created but not saved in the database.